File: fibonacci.py
def isFibonacci(n:int)->bool:
    a = 0 
    b = 1
    while a < n:
        a,b = b,a+b
    
    if a == n:
        return True
    return False

if __name__ == "__main__":
    n = int(input("Enter a number : "))

    if isFibonacci(n):
        print("Is a Fibonacci number.\n")
    else:
        print("Not a Fibonacci number.\n")

        

File: fibonacci.cpp
#include <iostream>
using namespace std;

int main() {
    int num;
    cout << "Enter a number: ";
    cin >> num;

    int a = 0, b = 1;

    while (a < num) {
        int c = a + b;
        a = b;
        b = c;
    }

    if (a == num)
        cout << " Is a Fibonacci Number";
    else
        cout << "Not a Fibonacci Number";

    return 0;
}
